pineappl_py/src/import_subgrid.rs

```diff
@@ -23,63 +24,28 @@ impl PyImportSubgridV1 {
     /// Parameters
     /// ----------
     /// array : numpy.ndarray(float)
-    ///     3D array with all weights
+    ///     `N`-dimensional array with all weights
     /// scales : list(float)
     ///     scales grid
-    /// x1_grid : list(float)
-    ///     first momentum fraction grid
-    /// x2_grid : Optional(list(float))
-    ///     second momentum fraction grid
+    /// x_grids : list(list(float))
+    ///     list with length `N` containing the momentum fractions (x1, x2, ...)
+    ///     which are also expressed as lists
     #[new]
     #[must_use]
-    pub fn new<'py>(
-        array: PyObject,
-        scales: Vec<f64>,
-        x1_grid: Vec<f64>,
-        x2_grid: Option<Vec<f64>>,
-        py: Python<'py>,
-    ) -> Self {
-        // let node_values: Vec<Vec<f64>> = vec![scales, x1_grid, x2_grid];
-        // let mut sparse_array: PackedArray<f64> =
-        //     PackedArray::new(node_values.iter().map(Vec::len).collect());
+    pub fn new(array: PyReadonlyArrayDyn<f64>, scales: Vec<f64>, x_grids: Vec<Vec<f64>>) -> Self {
```

Comment:
@Radonirinaunimi merge `scales` and `x_grids` in a single `node_values: Vec<Vec<f64>>`, because we may have more than two scales (flexible-scale fastNLO tables are an example).

Reply:
Addressed in [2a36294](https://github.com/NNPDF/pineappl/pull/299/commits/2a362946a3d7aee77475fa69b2596bb65fb9cf47).